picard/album.py

```diff
@@ -163,7 +163,21 @@ def _release_request_finished(self, document, http, error):
         self.load_task = None
         parsed = False
         try:
-            if error:
+            if (error
+                and config.setting['track_ars']
+                and not self.gateway_timeout
+                and repr(http.attribute(QtNetwork.QNetworkRequest.HttpStatusCodeAttribute)) == "504"):
+                # Gateway timeout error on large release with track relationships
+                # reissue as a priority request without the track relationships.
+                self.gateway_timeout = True
+                log.debug("Server timeout for release %r: retrying request without track relationships", self.id)
```

Comment:
Agree - should be log.warning.

In my experience, the timeout is consistent i.e. for a particular release it either always succeeds (i.e. the vast majority of releases) or it always fails - so Retry will not achieve anything.

We can still offer Retry without Track Relationships and Cancel - but what is the point?

However, I agree that a user dialog explaining what has been done would be a good idea. This would need to be done in the correct thread-aware way, and is beyond my skills.